[Doc] Release notes 7.5.1 (#11448)
elastic/logstash: docs/static/releasenotes.asciidoc

* <<logstash-7-0-0-alpha2,Logstash 7.0.0-alpha2>>
* <<logstash-7-0-0-alpha1,Logstash 7.0.0-alpha1>>

[[logstash-7-5-1]]
=== Logstash 7.5.1 Release Notes

> * Mention the path of DLQ to indicate DLQ if full for which pipeline https://github.com/elastic/logstash/pull/11280[#11280]

Reviewer: ```suggestion
* Improved usefullness of log messages when reporting full DLQ by including the relevant DLQ's path https://github.com/elastic/logstash/pull/11280[#11280]
```

> * Changed GemInstaller to don't blank  gemspec attribute https://github.com/elastic/logstash/pull/11340[#11340]

Reviewer: This changelog entry explains _what_ change happened, but doesn't show why a user should care about it.

```suggestion
* Fix: eliminates a crash that could occur at pipeline startup when the pipeline references a java-based plugin that had been installed via offline plugin pack https://github.com/elastic/logstash/pull/11340[#11340]
```

> * Support remove_field on metadata and tests [7.5 clean backport of #11334] https://github.com/elastic/logstash/pull/11342[#11342]

Reviewer: ```suggestion
* Fix: The common `remove_field` plugin option now correctly works on `@metadata` fields https://github.com/elastic/logstash/pull/11342[#11342]
```

* Fix: do not leak ThreadContext into the system https://github.com/elastic/logstash/pull/11356[#11356]

> * test codec against class name string to prevent class equivalence bug with a Delegator https://github.com/elastic/logstash/pull/11401[#11401]

Reviewer: Again, _what_, not _why_.

```suggestion
* Fix: eliminates a regression introduced in 7.2.0 where streaming-oriented inputs configured with payload-oriented codecs (`plain` or `json`) would use them as-is instead of using the appropriate line-oriented codec implementation (`lines` or `json_lines`, respectively) https://github.com/elastic/logstash/pull/11401[#11401]
```

> * update grok filter to 4.2.0 (for LS 7.5.1) https://github.com/elastic/logstash/pull/11432[#11432]

Reviewer: ```suggestion
* update grok filter to 4.2.0, adding `timeout_scope` option which can be used to reduce performance overhead https://github.com/elastic/logstash/pull/11432[#11432]
```

Reviewer: do we need to call this out here if we are already calling it out in the plugin changelog below?

* Fix: handle cloud-id with an empty kibana part https://github.com/elastic/logstash/pull/11435[#11435]
* bump dependencies for patch release https://github.com/elastic/logstash/pull/11438[#11438]

==== Plugins

*Dns Filter*

* Added documentation on the `nameserver` option for relying on `/etc/resolv.conf` to configure the resolver

*Elasticsearch Filter*

* Loosen restrictions on Elasticsearch gem https://github.com/logstash-plugins/logstash-filter-elasticsearch/pull/120[#120]

*Grok Filter*

> *  Added support for timeout_scope. 

Reviewer: This is effectively a duplicate as what is described in the next line

```suggestion
```

> Improved grok filter performance in multi-match scenarios. If you've noticed

Reviewer: does this bit need to be bullet-pointed to match formatting elsewhere in plugin notes?

Author: This is part of the previous bullet and renders correctly in the preview.
